### PR TITLE
fix mac_setup w/ tools+plotjuggler

### LINKED ISSUE #23800: Mac setup pycurl issue when import LogReader
`ImportError: pycurl: libcurl link-time ssl backends (secure-transport, openssl) do not include compile-time ssl backend (none/other)`

### PR DESCRIPTION
**Pull request by tecandrew**
**Description**: `tools/lib/robust_logreader.py` would error out when attempting to use `URLFile` which attempted to reference the wrong curl/openssl backend on macOS. compile flags and `curl` itself needed to be updated in order to properly use latest `pycurl` version

**Verification**: re-ran `./tools/mac_setup.sh` and ran `./plotjugger --demo`

image

**misc**: make setup script look nicer

🎯 #23458

### CODE REVIEW

File: tools/mac_setup.sh [hunk 1]
```diff
@@ -7,12 +7,13 @@ ROOT="$(cd $DIR/../ && pwd)"
 
 # Install brew if required
 if [[ $(command -v brew) == "" ]]; then
-  echo "Installing Hombrew"
+  echo "-- Installing Hombrew"
   /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
-  echo "[ ] installed brew t=$SECONDS"
+  echo "-- [✔] installed brew t=$SECONDS"
```
Comment (adeebshihadeh) [inline tools/mac_setup.sh:12]: can you revert all instances of this change?

File: tools/mac_setup.sh [hunk 2]
```diff
@@ -28,8 +29,8 @@ brew "libarchive"
 brew "libusb"
 brew "libtool"
 brew "llvm"
+brew "curl"
 brew "openssl"
-brew "pyenv"
```
Comment (adeebshihadeh) [inline tools/mac_setup.sh:32]: why was this removed? I think it's preferable to install with brew in this case

Comment (tecandrew) [reply tools/mac_setup.sh:32]: this version of `pyenv` oddly doesn't have the `update` command used in `update_requirements.sh` https://github.com/commaai/openpilot/blob/d5bdf1e951a2382b0df588b18183423d0ac26b56/update_requirements.sh#L18

Comment (adeebshihadeh) [reply tools/mac_setup.sh:32]: still want this to work on the version installed with brew

Comment (tecandrew) [reply tools/mac_setup.sh:32]: back in brew

**Comment by adeebshihadeh** [comment]
Still getting a pycurl error on my M1 MBP
```
macbookpro:openpilot adeeb$ pipenv shell
Loading .env environment variables...
Launching subshell in virtual environment...
bash-3.2$  . /Users/adeeb/.local/share/virtualenvs/openpilot-3HmyGjEJ/bin/activate
((openpilot) ) bash-3.2$ cd tools/plotjuggler/
((openpilot) ) bash-3.2$ ls
README.md               juggle.py               layouts                 test_plotjuggler.py
((openpilot) ) bash-3.2$ ./juggle.py --install
Traceback (most recent call last):
  File "./juggle.py", line 14, in <module>
    from selfdrive.test.process_replay.compare_logs import save_log
  File "/Users/adeeb/openpilot/selfdrive/test/process_replay/compare_logs.py", line 16, in <module>
    from tools.lib.logreader import LogReader
  File "/Users/adeeb/openpilot/tools/lib/logreader.py", line 8, in <module>
    from tools.lib.filereader import FileReader
  File "/Users/adeeb/openpilot/tools/lib/filereader.py", line 2, in <module>
    from tools.lib.url_file import URLFile
  File "/Users/adeeb/openpilot/tools/lib/url_file.py", line 8, in <module>
    import pycurl
ImportError: pycurl: libcurl link-time version (7.64.1) is older than compile-time version (7.81.0)
```

**Comment by adeebshihadeh** [comment]
Still getting a pycurl error on my M1 MBP
```
macbookpro:openpilot adeeb$ pipenv shell
Loading .env environment variables...
Launching subshell in virtual environment...
bash-3.2$  . /Users/adeeb/.local/share/virtualenvs/openpilot-3HmyGjEJ/bin/activate
((openpilot) ) bash-3.2$ cd tools/plotjuggler/
((openpilot) ) bash-3.2$ ls
README.md               juggle.py               layouts                 test_plotjuggler.py
((openpilot) ) bash-3.2$ ./juggle.py --install
Traceback (most recent call last):
  File "./juggle.py", line 14, in <module>
    from selfdrive.test.process_replay.compare_logs import save_log
  File "/Users/adeeb/openpilot/selfdrive/test/process_replay/compare_logs.py", line 16, in <module>
    from tools.lib.logreader import LogReader
  File "/Users/adeeb/openpilot/tools/lib/logreader.py", line 8, in <module>
    from tools.lib.filereader import FileReader
  File "/Users/adeeb/openpilot/tools/lib/filereader.py", line 2, in <module>
    from tools.lib.url_file import URLFile
  File "/Users/adeeb/openpilot/tools/lib/url_file.py", line 8, in <module>
    import pycurl
ImportError: pycurl: libcurl link-time version (7.64.1) is older than compile-time version (7.81.0)
```

**Comment by tecandrew** [comment]
that's weird. just did a fresh pull of OP and full reinstall of `brew` and encountered no issues. might be that OP is still referencing the xcode sdk version of `curl`? (outdated from the perspective of the newer `pip pycurl`).

**Comment by pd0wm** [comment]
I had to set `export PYCURL_SSL_LIBRARY=openssl` before installing pycurl. Does that help in your case?

**Comment by pd0wm** [comment]
Getting a similar error as @adeebshihadeh `ImportError: pycurl: libcurl link-time version (7.77.0) is older than compile-time version (7.81.0)`

**Comment by pd0wm** [comment]
Using system curl, but openssl from homebrew fixes things for me. Let me know if this causes issues on non-M1 machines.

**Pull request merged**